### PR TITLE
fix(unity): Initialization and programmatic configuration

### PR DESCRIPTION
**Pull request by bitsandfoxes**
Updated the parts explaining how the SDKs get initialized and what limitations this imposes on mobile.

### CODE REVIEW

File: src/includes/configuration/config-intro/unity.mdx [hunk 1]
```diff
@@ -1,8 +1,12 @@
...
 
-By default, we initialize the native SDKs before the Unity engine itself. That means it also runs before the C# layer that relies on [BeforeSceneLoad RuntimeInitializeOnLoadMethodAttribute](https://docs.unity3d.com/ScriptReference/RuntimeInitializeLoadType.BeforeSceneLoad.html) to get initialized. This allows us to capture bugs/crashes of the engine itself, any native plugin, and anything written in C#. The setup for the native SDKs happens during build time, which is why we rely on the options being set in the Sentry editor configuration window and saved to `Assets/Resources/Sentry/SentryOptions.asset`. To provide a way to modify options programmatically, we've added `ScriptableOptionsConfiguration` to the `Options Config` tab in the Sentry editor window.
+The C# layer self-initializes through the use of the [BeforeSceneLoad RuntimeInitializeOnLoadMethodAttribute](https://docs.unity3d.com/ScriptReference/RuntimeInitializeLoadType.BeforeSceneLoad.html). On Desktop, the C# layer is also responsible for intializing the native SDKs.
```
Comment (imatwawana) [inline src/includes/configuration/config-intro/unity.mdx:5]: I don't think "Desktop" needs to be capitalized here?

```suggestion
The C# layer self-initializes through the use of the [BeforeSceneLoad RuntimeInitializeOnLoadMethodAttribute](https://docs.unity3d.com/ScriptReference/RuntimeInitializeLoadType.BeforeSceneLoad.html). On desktop, the C# layer is also responsible for intializing the native SDKs.
```

File: src/includes/configuration/config-intro/unity.mdx [hunk 2]
```diff
@@ -1,8 +1,12 @@
...
+
+On mobile, the native SDKs are configured and set up during build time and initialize themselves before the Unity engine. This allows us to capture bugs/crashes of the engine itself. For that reason we rely on on the options being set in the Sentry editor configuration window and saved to `Assets/Resources/Sentry/SentryOptions.asset`.
```
Comment (imatwawana) [inline src/includes/configuration/config-intro/unity.mdx:7]:
```suggestion
On mobile, the native SDKs are configured and set up during build time and initialize themselves before the Unity engine. This allows us to capture bugs/crashes of the engine itself. For that reason, we rely on on the options being set in the Sentry editor configuration window and saved to `Assets/Resources/Sentry/SentryOptions.asset`.
```

**Review by imatwawana — APPROVED** [review]
Just a couple wording things

**Comment by bitsandfoxes** [comment]
> Just a couple wording things

Thank you as always!

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**